embedding/embedlite/embedshared/nsWindow.cpp

```diff
@@ -276,11 +285,57 @@ layers::LayersId nsWindow::GetRootLayerId() const
   return mCompositorSession ? mCompositorSession->RootLayerTreeId() : layers::LayersId{0};
 }
 
-void nsWindow::SetContentController(mozilla::layers::GeckoContentController *aController)
+#ifdef MOZ_LOGGING
+static void LogControllerStack(std::list<EmbedContentController *> &controllers)
+{
+  int pos = 0;
+  for (std::list<EmbedContentController *>::const_iterator it = controllers.begin(); it != controllers.end(); ++it) {
+    LOGT("Stack controller %d : %u", pos, (*it)->GetUniqueID());
+    ++pos;
+  }
+}
+#endif // MOZ_LOGGING
+
+void nsWindow::SetContentController(EmbedContentController *aController)
 {
   if (mCompositorSession) {
     mCompositorSession->SetContentController(aController);
   }
+
+  MOZ_ASSERT(aController);
+
+  // If the view is deactivated we need to reactivate the previous view's
+  // content controller, so we store the controllwers on a stack
+  if (mControllers.empty() || aController != mControllers.back()) {
```

Comment:
```suggestion
  if (mControllers.empty() || aController->GetUniqueID() != mControllers.back()->GetUniqueID()) {
```

Could it happen that we have multiple instances of the same content controller in the stack? Wondering if we should always move the existing instance to the top. Surely, that is not needed if there cannot be multiple instances.